tests/util/test_yaml_extension.cpp

```diff
@@ -4,29 +4,62 @@
 //==============================================================================
 TEST(YamlEigenExtension, Errors)
 {
-  std::string notSequential = "name: PRL";
-  EXPECT_THROW(YAML::Load(notSequential)["name"].as<Eigen::VectorXd>(),
+  using Vector1d = Eigen::Matrix<double, 1, 1>;
+
+  std::string empty = "name:";
+  EXPECT_THROW(YAML::Load(empty)["name"].as<Eigen::VectorXd>(),
+      YAML::RepresentationException);
+
+  std::string notSequential1 = "name: PRL";
+  EXPECT_THROW(YAML::Load(notSequential1)["name"].as<Eigen::VectorXd>(),
+      YAML::RepresentationException);
+
+  std::string null = "name: []";
+  EXPECT_THROW(YAML::Load(null)["name"].as<Eigen::VectorXd>(),
+      YAML::RepresentationException);
+
+  std::string map = "status: {left: ok, right: bad}";
+  EXPECT_THROW(YAML::Load(map)["status"].as<Eigen::VectorXd>(),
+      YAML::RepresentationException);
+
+  std::string scalar = "scalar: 1";
+  EXPECT_THROW(YAML::Load(scalar)["scalar"].as<Vector1d>(),
+      YAML::RepresentationException);
+
+  std::string vector1 = "vector: [1]";
+  EXPECT_NO_THROW(YAML::Load(vector1)["vector"].as<Vector1d>());
+
+  std::string incorrectSize1 = "vector: [1, 2, 3]";
+  EXPECT_THROW(YAML::Load(incorrectSize1)["vector"].as<Eigen::Vector2d>(),
+      YAML::RepresentationException);
+
+  std::string incorrectSize2 = "matrix: [[1, 2], [3, 4], [5, 6]]";
+  EXPECT_THROW(YAML::Load(incorrectSize2)["matrix"].as<Eigen::Matrix3d>(),
+      YAML::RepresentationException);
+
+  std::string notMatrix1 = "matrix: [[1, 2, 3], [1, 2, 3], 1]";
+  EXPECT_THROW(YAML::Load(notMatrix1)["matrix"].as<Eigen::Matrix3d>(),
       YAML::RepresentationException);
```

Comment:
Add the converse of this test, which starts with a scalar and has a `Sequence` as a later element. This is an interesting test case because we use `node[0].Type()` to decide whether we're deserializing a matrix or a vector.